src/org/zaproxy/zap/extension/websocket/WebSocketAPI.java

```diff
@@ -0,0 +1,444 @@
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+import org.zaproxy.zap.extension.api.ApiView;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy.Initiator;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy.State;
+import org.zaproxy.zap.extension.websocket.utility.InvalidUtf8Exception;
+import org.zaproxy.zap.extension.websocket.utility.WebSocketUtils;
+import org.zaproxy.zap.model.StructuralNode;
+import org.zaproxy.zap.model.Target;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+public class WebSocketAPI extends ApiImplementor {
+
+    private static final String PREFIX = "websocket";
+
+    private ExtensionWebSocket extension;
+
+    private static final String VIEW_CHANNELS = "channels";
+    private static final String VIEW_MESSAGE = "message";
+    private static final String VIEW_MESSAGES = "messages";
+
+    private static final String ACTION_SEND_TEXT_MESSAGE = "sendTextMessage";
+
+    private static final String PARAM_COUNT = "count";
+    private static final String PARAM_START = "start";
+    private static final String PARAM_PAYLOAD_PREVIEW_LENGTH = "payloadPreviewLength";
+    private static final String PARAM_CHANNEL_ID = "channelId";
+    private static final String PARAM_MESSAGE_ID = "messageId";
+    private static final String PARAM_OUTGOING = "outgoing";
+    private static final String PARAM_MESSAGE = "message";
+
+    private static Logger LOG = Logger.getLogger(WebSocketAPI.class);
+
+    private WebSocketObserver observer;
+
+    private String callbackUrl;
+
+    public WebSocketAPI(ExtensionWebSocket extension) {
+        this.extension = extension;
+
+        this.addApiView(new ApiView(VIEW_CHANNELS));
+        this.addApiView(new ApiView(VIEW_MESSAGE, new String[] { PARAM_CHANNEL_ID, PARAM_MESSAGE_ID }));
+        this.addApiView(
+                new ApiView(
+                        VIEW_MESSAGES,
+                        null,
+                        new String[] { PARAM_CHANNEL_ID, PARAM_START, PARAM_COUNT, PARAM_PAYLOAD_PREVIEW_LENGTH }));
+
+        this.addApiAction(
+                new ApiAction(ACTION_SEND_TEXT_MESSAGE, new String[] { PARAM_CHANNEL_ID, PARAM_OUTGOING, PARAM_MESSAGE }));
+
+        callbackUrl = API.getInstance().getCallBackUrl(this, "https://" + API.API_DOMAIN);
+
+    }
+
+    public String getCallbackUrl() {
+        return this.callbackUrl.replace("https://", "wss://");
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public String handleCallBack(HttpMessage msg) throws ApiException {

```

Comment:
For consistency drop the cap on Back?

Reply:
Not possible here/now (overriding core class).

Reply:
K